File: bharat-railway-backend/src/main/java/com/bharatrailway/shared/exception/ErrorResponse.java
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-backend
 * Branch: feature/backend-developer-hitanshu
 * Developer: Hitanshu Dhakrey
 * Assisted by: Chandra Shekhar Bansal (Infrastructure), DeepSeek (AI Scribe)
 * Date: 2026-06-19
 * Version: 0.1.0-SNAPSHOT
 *
 * Description:
 * Standardized JSON error response DTO.
 * Every exception maps to this single contract for consistent client handling.
 */

package com.bharatrailway.shared.exception;

import java.time.OffsetDateTime;

import com.fasterxml.jackson.annotation.JsonFormat;

public record ErrorResponse(
        @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ssXXX")
        OffsetDateTime timestamp,
        int status,
        String error,
        String message,
        String path
) {}

File: bharat-railway-frontend/lib/models/responses/error_response.dart
/*
 * Project: Bharat Railway Booking System
 * Module: bharat-railway-frontend
 * Branch: feature/frontend-developer-chandrashekhar
 * Developer: Chandra Shekhar Bansal
 * Assisted by:
 * Date: 2026-07-06
 * Version: 1.0.0
 *
 * Description:
 * Error response DTO for the authentication API.
 * Maps to backend ErrorResponse.java record.
 * Used globally for all API error responses.
 * Fields: timestamp, status, error, message, path.
 */

class ErrorResponse {
  final String timestamp;
  final int status;
  final String error;
  final String message;
  final String path;

  ErrorResponse({
    required this.timestamp,
    required this.status,
    required this.error,
    required this.message,
    required this.path,
  });

  factory ErrorResponse.fromJson(Map<String, dynamic> json) {
    return ErrorResponse(
      timestamp: json['timestamp'] as String,
      status: json['status'] as int,
      error: json['error'] as String,
      message: json['message'] as String,
      path: json['path'] as String,
    );
  }
}